train.py

```diff
@@ -456,58 +372,60 @@ def eval(
             metric_tracker_val.save()
 
             ################################################################################ Saving models
-            if args.save:
-                state = {
-                    "args": args,
-                    "epoch": epoch,
-                    "model": model.state_dict(),
-                    "optimizer": optimizer.state_dict(),
-                    "scheduler": scheduler.state_dict(),
-                }
-
-                current_epoch_filename = f"{epoch}_ckpt.pth.tar"
-                latest_epoch_filename = "latest_ckpt.pth.tar"
-
-                save_checkpoint(
-                    state=state,
-                    directory=saved_models_filepath,
-                    filename=current_epoch_filename,
-                    is_best=False,
-                )
 
-                save_checkpoint(
-                    state=state,
-                    directory=saved_models_filepath,
-                    filename=latest_epoch_filename,
-                    is_best=False,
-                )
-            ############################################################################################################
+            state = {
+                "args": args,
+                "epoch": epoch,
+                "model": model.state_dict(),
+                "optimizer": optimizer.state_dict(),
+                "scheduler": scheduler.state_dict(),
+            }
+
+            metric_tracker_val.refresh_best_n_epoch_models(
+                directory=saved_models_filepath,
+                filename=args.experiment_name,
+                metric_name="accuracy",
+                n=args.save_top_n_val_models,
+                bigger_is_better=True,
+                current_epoch_idx=epoch,
+                current_epoch_state=state,
+            )
+
+            save_checkpoint(
+                state=state,
+                directory=saved_models_filepath,
+                filename=args.experiment_name,
+                is_best=False,
+            )
+        ############################################################################################################
 
         if args.test:
             test_progress_dict = {
                 **dict(description="[green]Training", total=len(test_set_loader)),
                 **metric_tracker_test.get_current_iteration_metric_text_column_fields(),
             }
 
-            test_epoch_progress = epoch_progress.add_task(**test_progress_dict)
+            test_epoch_progress = progress_reporter.epoch_progress.add_task(
+                **test_progress_dict
```

Comment:
why do you need this here and not for the main training loop?

Reply:
It avoids having 'testing' in the terminal interface before it is necessary. :smile: